src/test/ui/option-unwrap.rs

```diff
@@ -1,32 +0,0 @@
-// run-pass
-
-#![allow(non_camel_case_types)]
-use std::cell::Cell;
-
-struct dtor<'a> {
-    x: &'a Cell<isize>,
-}
-
-impl<'a> Drop for dtor<'a> {
-    fn drop(&mut self) {
-        self.x.set(self.x.get() - 1);
-    }
-}
-
-fn unwrap<T>(o: Option<T>) -> T {
-    match o {
-      Some(v) => v,
-      None => panic!()
-    }
-}
-
-pub fn main() {
-    let x = &Cell::new(1);
-
-    {
-        let b = Some(dtor { x:x });
-        let _c = unwrap(b);
-    }
-
-    assert_eq!(x.get(), 0);
-}
```

Comment:
This test feels like it intends to stresses auto-generated drop impl, rather than any specific impl details of standard library, so I think it's OK to keep as an UI test.  The best way to check if this intends to check compiler or stdlib is to look at the PR which introduced the test: https://github.com/rust-lang/rust/commit/e47d2f60607142eaf68a0c560f5c3d37fd1da13d#diff-518b4158ed084dab26591f1ec09dffef. In this case, this is a change to the compiler.

But keeping this as a `#[test]` sounds fine by me as well!

Reply:
I don't have strong feelings, I just felt it was simple enough to move. I can move it back if necessary.